src/codegen/optimizer.lisp

```diff
@@ -24,21 +27,57 @@
    #:make-candidate-manager)
   (:import-from
    #:coalton-impl/codegen/transformations
-   #:traverse-bindings
+   #:action
    #:traverse
-   #:update-function-env
-   #:make-function-table)
+   #:traverse-with-binding-list)
+  (:import-from
+   #:coalton-impl/codegen/transformations
+   #:node-free-p)
   (:local-nicknames
    (#:settings #:coalton-impl/settings)
    (#:util #:coalton-impl/util)
    (#:parser #:coalton-impl/parser)
    (#:tc #:coalton-impl/typechecker))
   (:export
+   #:make-function-table
    #:optimize-bindings
    #:optimize-node))
 
 (in-package #:coalton-impl/codegen/optimizer)
 
+(defun update-function-env (bindings env)
```

Comment:
I brought these two functions out of `transformations` to the place they are used, in `optimizer`, since they don't seem like transformations to me, but if that's wrong I can put them back.